apps/example-solid-start/src/routes/index.tsx

```diff
@@ -0,0 +1,27 @@
+import { type ParentComponent } from "solid-js";
+import { Title, useRouteData } from "solid-start";
+import { createServerData$ } from "solid-start/server";
+import { authOpts } from "./api/auth/[...solidauth]";
+import { getSession } from "@solid-auth/next/session";
+
+export const routeData = () => {
+  return createServerData$(
+    async (_, { request }) => {
+      return await getSession(request, authOpts);
+    },
+    { key: () => ["auth_user"] }
+  );
+};
```

Comment:
This looks a but messy, could we abstract away more in the package?

Reply:
i could create a createServerSession hook that takes authOpts, and will return a useServerSession hook that can be used directly in the routeData.

should i?

Reply:
this is how the solid routeData looks, routeData is like getServerSideProps, key is not required and createServerData$ just tells it to run it on the server, it is very similar to nextjs